tests/network/vmi_istio.go

```diff
@@ -0,0 +1,313 @@
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+
+			By("Waiting for VMI to be ready")
+			tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+		})
+		Describe("Inbound traffic", func() {
+			var job *batchv1.Job
+			table.DescribeTable("request to VMI", func(port int) {
+				job, err = createJobCheckingVMIReachability(vmi, port)
+
+				By("Waiting for the job to succeed")
+				Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).To(Succeed())
+			},
+				table.Entry("should reach VMI HTTP server on service declared port", svcDeclaredTestPort),
+				table.Entry("should reach VMI HTTP server on service undeclared port", svcUndeclaredTestPort),
+			)
+			Context("With PeerAuthentication enforcing mTLS", func() {
+				peerAuthenticationRes := schema.GroupVersionResource{Group: "security.istio.io", Version: "v1beta1", Resource: "peerauthentications"}
+				var peerAuthentication *unstructured.Unstructured
+				BeforeEach(func() {
+					peerAuthentication = &unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"apiVersion": "security.istio.io/v1beta1",
+							"kind":       "PeerAuthentication",
+							"metadata": map[string]interface{}{
+								"name": "strict-pa",
+							},
+							"spec": map[string]interface{}{
+								"mtls": map[string]interface{}{
+									"mode": "STRICT",
+								},
+							},
+						},
+					}
+					_, err = virtClient.DynamicClient().Resource(peerAuthenticationRes).Namespace(tests.NamespaceTestDefault).Create(context.Background(), peerAuthentication, metav1.CreateOptions{})
+					ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+				})
+				table.DescribeTable("request to VMI", func(port int) {
+					job, err = createJobCheckingVMIReachability(vmi, port)
+
+					By("Waiting for the job to fail")
+					Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).NotTo(Succeed())
+
+					err = virtClient.DynamicClient().Resource(peerAuthenticationRes).Namespace(tests.NamespaceTestDefault).Delete(context.Background(), peerAuthentication.GetName(), metav1.DeleteOptions{})
+					ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+				},
+					table.Entry("client outside mesh should not reach VMI HTTP server on service declared port", svcDeclaredTestPort),
+					table.Entry("client outside mesh should not reach VMI HTTP server on service undeclared port", svcUndeclaredTestPort),
+				)
+			})
+		})
+		Describe("Outbound traffic", func() {
+			var (
+				serverVMIAddress string
+				serverVMI        *v1.VirtualMachineInstance
+				testPort         = 4200
+				sidecarRes       = schema.GroupVersionResource{Group: "networking.istio.io", Version: "v1beta1", Resource: "sidecars"}
+			)
+
+			curlCommand := func(serverIP string, port int) string {
+				return fmt.Sprintf("curl -sD - -o /dev/null http://%s:%d | head -n 1\n", serverIP, port)
+			}
+
+			generateExpectedHTTPReturnCodesRegex := func(shouldBeReachable bool) (string, error) {
```

Comment:
Having a boolean argument is not a good thing; it's a clear indication your function does 2 things.

Wouldn't it simplify your code if you had 2 regexes ?

Reply:
Yeah this wasn't good. I went ahead to make 2 regexes, but then I didn't like that it's only the code that's different.
So I've parametrized it by the return code instead. Should make more sense  this way